src/components/CIPipelineN/CustomImageTags.tsx

```diff
@@ -0,0 +1,230 @@
+    const [showImageTagPatternDetails, setShowImageTagPatternDetails] = useState<boolean>(false)
+    const isCustomTagError = formDataErrorObj.customTag.message.length > 0 && !formDataErrorObj.customTag.isValid
+
+    const toggleEditAction = () => {
+        setShowImageTagPatternDetails(!showImageTagPatternDetails)
+    }
+
+    const handleImageTagTypeChange = (event) => {
+        setImageTagValue(event.target.value)
+    }
+
+    const renderInputErrorMessage = (errorMessage: string) => {
+        return (
+            <span className="flexbox cr-5 mt-4 fw-5 fs-11 flexbox">
+                <AlertTriangle className="icon-dim-14 mr-5 ml-5 mt-2" />
+                <span>{errorMessage}</span>
+            </span>
+        )
+    }
+
+    const renderImageTagPatternPreview = () => {
+        return (
+            <div className="pl-16 pr-16 pt-12 pb-12">
+                <RadioGroup
+                    value={imageTagValue}
+                    name="image-type"
+                    onChange={handleImageTagTypeChange}
+                    className="chartrepo-type__radio-group dc__block-imp"
+                >
+                    <RadioGroupItem dataTestId="ci-default-image-tag" value={ImageTagType.Default}>
+                        Use system generated image tag pattern
+                        {imageTagValue === ImageTagType.Default && getDefaultTagValue()}
+                    </RadioGroupItem>
+                    <br />
+                    <RadioGroupItem dataTestId="ci-custom-image-tag" value={ImageTagType.Custom}>
+                        Create custom image tag pattern
+                    </RadioGroupItem>
+                </RadioGroup>
+                {renderCustomImageDetails()}
+            </div>
+        )
+    }
+
+    const onChangeCustomInput = (event) => {
+        const _form = { ...formData }
+        _form.customTag = {
+            ..._form.customTag,
+            tagPattern: event.target.value,
+        }
+        setFormData(_form)
+        const _formDataErrorObj = { ...formDataErrorObj }
+        _formDataErrorObj.customTag = validationRules.customTag(event.target.value)
+        setFormDataErrorObj(_formDataErrorObj)
+    }
+
+    const onChangeCustomImageCounter = (event) => {
+        const _form = { ...formData }
+        _form.customTag = {
+            ..._form.customTag,
+            counterX: +event.target.value,
+        }
+        setFormData(_form)
+    }
+
+    const renderCustomImageDetails = () => {
+        return (
+            imageTagValue === ImageTagType.Custom && (
+                <div className="pl-26">
+                    <span className="cn-7">Use mix of fixed pattern and variable {`{x}`}</span>
+                    <textarea
+                        tabIndex={1}
+                        className="form__input form__input-no-bottom-radius"
+                        placeholder="Example: v1.2.{x}"
+                        name="image_tag"
+                        autoComplete={'off'}
```

Comment:
it should be written as autoComplete="off"

Reply:
Done